api/types.go

```diff
@@ -190,6 +198,8 @@ type ChatResponse struct {
 	Message    Message   `json:"message"`
 	DoneReason string    `json:"done_reason,omitempty"`
 
+	Debug map[string]any `json:"debug,omitempty"`
```

Comment:
The name `debug` isn't a good choice here since we aren't really de-bugging something as much as looking to have the final prompt echo'd back

Reply:
@ParthSareen Please be more judicious with your uses of `map[string]any`. That is to say, it should be so rare you forget it is an option most of the time.

To implement the proposal, which states this is an object, always, use a `struct` with fields of specific types, not `any`.

This also allows for docs, which we need much more of in ollama.

```
    //...

    // Debug, if set, adds additional context to generate responses as
    // specified by its fields.
    Debug: *DebugOptions,
}

// DebugOptions defines options available for generate requests.
type DebugOptions struct {
    // Mode specifies which mode of debugging is requested of the server. The
    // available modes are "default", and "prompt. If empty, "default" is used.
    Mode string `json:"mode"`
}

```